engine/src/flutter/lib/web_ui/lib/src/engine/util.dart

```diff
@@ -551,6 +551,52 @@ bool listEquals<T>(List<T>? a, List<T>? b) {
   return true;
 }
 
+/// Determines if lists [a] and [b] are deep equivalent, regardless of their
+/// order.
+///
+/// Returns true if the lists are both null, or if they are both non-null, have
+/// the same length, and contain the same elements regardless of their order.
+/// Returns false otherwise.
+bool unorderedListEqual<T>(List<T>? a, List<T>? b) {
+  if (a == b) {
+    return true;
+  }
+  if ((a == null) != (b == null)) {
+    return false;
+  }
+  // They most both be non-null now.
+  if (a!.length != b!.length) {
```

Comment:
Are `null` and empty list not equivalent?

Reply:
will update